### DISCUSSION TITLE
new dtb for 905W2 4gb

### DISCUSSION THREAD

**Discussion by ghost** [devmfc/debian-on-amlogic#76] [Q&A] [answered]
Firts, THANK YOU for your great work. 
I know support for 905w2 is absolutely experimental, but...is possible to have a dtb for 4gb ram version?
i've got a unlabeled 905w2 4/64 (with generic 905w2 i see only 32gb emmc but is not a real problem...but only 1,8gb ram! i know some amount of ram is reserved but...they have 4 xD)

btw, for my setup (server mysql mariadb) maybe is ok 2gb but..well box have 4, why don't use it? :-P LoL 

so, basically, can we have a "generic 905w2" but with 4gb support or simply, or, how can i edit the 201 to get my 4gb working (i only know how to get dts file but...next step is obscure...)? tnx in advance

**Comment by ghost** [comment]
btw, in my clone of clone of maybe another clone box:
(https://www.amazon.it/dp/B0D78FXTWJ this is my box)

**_What Works:_**

LAN 100mb
Wi-Fi 2.4/5
HDMI output
USB 2.0
write boot entryes with _"./aml-multiboot-setup.sh"_
write to emmc with _"./aml-install-to-emmc.sh"_ (i think, 'couse there was no error in compile and the latest response is "now reboot or not etc", i reboot and i don't need to use anymore toothpick for start)

**_What Not works:_**

ram (say me 2GB instead of 4GB)
emmc capacity (say me 32 instead 64...but only if i list hardware...in main page at start only see sda/sda1 (usb dongle))
SD (didn't detected...don't know how to solve...i'll prefer to boot with this instead USB due to low space on box tv stand but this is a chimera...)
USB 3.0
Optical output and audio

nothing others.

ouch yep, my english did not works ahahah (pls sorry 4 my bad eng..is not my main lang)

**Comment by devmfc** [comment, marked as answer]
Thanks, you're welcome!

I'm afraid that there is a high chance you bought a box with fake specifications. eMMC has no DTB settings to specify the size. It is not needed, because you can just ask eMMC its size. So that is not eMMC module or driver or dtb or even bootloader specific.

The memory size node is not present in the meson-s4-s905w2-ap201.dtb (generic), so the bootloader is respnsible to set the right size. So again, it is not the DTB that specifies the memory size. And it is still 2GB....   So to get the real size of memory/eMMC you should read the output of the serial output of the bootloader or to open the box and google for the specifications of the eMMC chip and memory chips.

**Comment by ghost** [comment]
Thank you...i'll switch to s905x3 and all works great :D working emmc (32gb) 4gb ram (well ok 3,x) and all perfect! :D i think the previous is scam/fake board..resend back to amazon and buy T95max+ yesterbay...today up and running! ('bout 20min ago ahahah) LoL so, thanx for all your work hun! :D cheers!

**Reply by danidanibaerboo** [reply]
this pretty much what happen to me previously, but in my case they tamper with the Uboot too, making any flash tool unusable and thus bricked.

**Reply by ghost** [reply]
Reall... is a pain ahahah ...btw i'll return it to amazon and bui a T95 max+... setup  with tanix tx3 dtb and works great!

![Screenshot 2024-08-09 065323](https://github.com/user-attachments/assets/a94e9486-0435-410d-beeb-ae33a1256ce0)
